### PR TITLE
DAOS-13521 control: Ignore config on daos_server nvme reset

### PR DESCRIPTION
**Pull request by tanabarr**
Change behaviour of daos_server nvme reset command so that any server
config file is ignored. The main effect of this is that the 
spdk/scripts/setup.sh reset call is made without assigning any
PCI_ALLOWED filter. The result being that all VMD domains are reset
even if the backing devices have no driver bindings. This fix should
resolve scenarios related to https://github.com/spdk/spdk/issues/2926.

The change shouldn't  cause any undesired side effects as DAOS is
assumed to be the only SPDK consumer on a host and so returning all
NVMe devices to the kernel device drivers after use is acceptable.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by knard-intel — APPROVED** [review]
Seems OK for me

File: src/control/cmd/daos_server/storage_nvme.go [hunk 1]
```diff
@@ -253,11 +256,13 @@ func (cmd *prepareNVMeCmd) Execute(_ []string) error {
...
 
 type resetNVMeCmd struct {
-	nvmeCmd
-	PCIBlockList string `long:"pci-block-list" description:"Comma-separated list of PCI devices (by address) to be ignored when unbinding devices from Kernel driver to be used with SPDK (default is no PCI devices)"`
-	TargetUser   string `short:"u" long:"target-user" description:"User that will own hugepage mountpoint directory and vfio groups."`
-	DisableVFIO  bool   `long:"disable-vfio" description:"Force SPDK to use the UIO driver for NVMe device access"`
-	Args         struct {
+	cmdutil.LogCmd  `json:"-"`
+	helperLogCmd    `json:"-"`
+	iommuCheckerCmd `json:"-"`
```
Comment (kjacque) [inline src/control/cmd/daos_server/storage_nvme.go:261]: Are the json annotations needed? I don't think we serialize command structs to json anywhere.

**Review by kjacque — APPROVED** [review]
Minor comment, otherwise looks okay to me.

**Review by mjmac — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Comment by tanabarr** [comment]
replaced by a different solution: https://github.com/daos-stack/daos/pull/12397

**Pull request closed without merging**